components/server/src/server.rs

```diff
@@ -1874,9 +1877,27 @@ impl EnginesResourceInfo {
         if let Some(raft_engine) = &self.raft_engine {
             fetch_engine_cf(raft_engine, CF_DEFAULT, &mut normalized_pending_bytes);
         }
-        for cf in &[CF_DEFAULT, CF_WRITE, CF_LOCK] {
-            fetch_engine_cf(&self.kv_engine, cf, &mut normalized_pending_bytes);
+
+        // tablets record the tablet with the latest suffix in each region
+        let mut tabltes = HashMap::<u64, (u64, RocksEngine)>::new();
+        self.tablet_factory.for_each_opened_tablet(
+            &mut |id, suffix, db: &RocksEngine| match tabltes.entry(id) {
+                collections::HashMapEntry::Occupied(mut slot) => {
+                    if slot.get().0 < suffix {
+                        slot.insert((suffix, db.clone()));
+                    }
+                }
+                collections::HashMapEntry::Vacant(slot) => {
+                    slot.insert((suffix, db.clone()));
+                }
+            },
+        );
+        for (_, (_, tablet)) in tabltes {
+            for cf in &[CF_DEFAULT, CF_WRITE, CF_LOCK] {
+                fetch_engine_cf(&tablet, cf, &mut normalized_pending_bytes);
```

Comment:
is it possible that the tablet can be destroyed while being referenced here?
I think it's probably fine here if we won't destroy the tablet once the region peer is moved to other TiKV, otherwise there's race condition here that once for_each_opened_tablet is done,  some tablet is deleted already. 
To make it more robust,  I think we may add a flag in TabletFactory to mark it being used so that the data files or the underlying memory structure won't be deleted.

Reply:
It indeed has the race condition problem. Let me come up with a way to handle with it.